src/const-eval.md

```diff
@@ -0,0 +1,32 @@
+# Constant Evaluation
```

Comment:
It would be good to mention where in the compiler "pipeline" const eval happens. IIUC, it is after MIR is constructed and borrowck?

Reply:
Note that the "pipeline" model is no longer entirely accurate. We care more about *dependencies* - and yes, `miri` requires MIR to be constructed to run on, but e.g. the type-checking, MIR construction and MIR evaluation of a constant might happen *before* any of those ever happen on a different constant/function/etc.

We *have to* to do all that to understand even the type `[T; 8]` - type-checking and working MIR is therefore reentrant.

Reply:
> Note that the "pipeline" model is no longer entirely accurate.

Yep, that's why I put it in "quotes" 😛 What I was getting at more was what dependencies const eval has, as you noted.